decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDeciderCli.java

```diff
@@ -287,7 +294,7 @@ public ReturnValue init() {
         }
         decider.setExcludeFilters(excludeFilters);
 
-        if (!(options.has(allOpt) ^ !includeFilters.isEmpty() ^ !excludeFilters.isEmpty())) {
+        if (!(options.has(allOpt) ^ (!includeFilters.isEmpty() || !excludeFilters.isEmpty()))) {
```

Comment:
I had to squint at this for a long time. if --all is not specified and neither of the filters is empty this will be true. I don't think you intend this?